### ISSUE TITLE
OPERATION GET RID OF .rscollection OR AT LEAST HAVE AN ALTERNATIVE

### ISSUE THREAD

**Issue by felisucoibi** [RetroShare/RetroShare#635]
2017-09 I update the url structure....

OPERATION GET RID OF .rscollection OR AT LEAST HAVE AN ALTERNATIVE

In retroshare there is kind of .torrent files, that needs to be donewloaded before being processed.

A .rscollection at this moment is a complicated problem for user to share because of:
1. Yo need to create a .rscollection file
2. You have to share it in a shared folder
3. You have to wait till is hashed
4. You have to download it
5. You have to open it (and there are some rs bugs related to it)
6. You have to acept directories etc...

Why NOT we just add the directory information of a file in the same rslink??

At this moment a rslink to a file looks like this:
retroshare://file?name=Frontends.rar&size=628027003&hash=bbebbc45214c3d8b14aa73ac3e721cff2c3f647c

Just adding a little more info to the link we have a directory structure:

retroshare://file?name=Frontends.rar&size=628027003&hash=bbebbc45214c3d8b14aa73ac3e721cff2c3f647c&directoryname=dir1/dir2

In this case it will create this inside the default download directory of RS:
/dir1/dir2/Frontends.rar

So you can share a folder as a RSLINK with the files inside without creating a .rscollection file, without uploading to a shareddirectory, without the download process etc....

WIN WIN

Possible problems:

Security, boviously like should be done with filenames url should be fileterd.
1. Not allow ../ stuff etc
2. Not allow strange directories structures incompatible with windows or linux
3. other security considerations

That are the same or should be the same as filenames, don't know wich filters are used atm with a file name, i heard there are problem with long file names, or incompatible chars between linux-windows.

BOUNTY: There are 100€ waiting for somebody to code it.

A: It should work at least when dragging a folder in attached files of CHANNELS in a CHANNEL post, so when dragging the directory there the strcuture is respected in the rs links so when download the files of the channels the directory name is created.
B: Also should work in file lists, own or friends ONES, left button, "copy rs link with directory structure" so it creates the file links inside with the directory information in the link and copy them to clipboard, like when selecting some files and left button copy rs link, but with directory strcuture.

Obviously also when downloading any of these links with the directory info, the directory mas be created etc....

**Comment by G10h4ck** [comment]
I have been looking at rscollection thing, and it seems to me that we may eventually use .torrent files instead or support both, this would ease inter-operation too as if someone find a .torrent in the internet could try to download it  via retroshare instead of having no other option then to use a torrent client, another thing that would be nice is support for magnet links

**Comment by cavebeat** [comment]
There is a Bounty waiting to get earned for solving of this issue [![Bountysource](https://api.bountysource.com/badge/issue?issue_id=40779372)](https://www.bountysource.com/issues/40779372-operation-get-rid-of-rscollection-or-at-least-have-an-alternative?utm_source=40779372&utm_medium=shield&utm_campaign=ISSUE_BADGE)

**Comment by felisucoibi** [comment]
Nice idea to use .torrents so it can even work on both sides, but my proposal still is waiting to somebody to code, no files, just all the info inside the rs linki, directory structure and file and hash.... all in the same link...

**Comment by felisucoibi** [comment]
I updated the info for the bounty and the definition of the structure.

**Comment by ASmith-** [comment]
The Bounty states $100 for your, cave's desire to do away with the existing .rscollection file coding which I might add currently there are thousands of pre-existing .rscollection files in the Retroshare File Sharing Network. Towards the end of this pay to program post of caves or yourself, it states 100 Euros using the Euro Symbol which is about 1.3x more than $100 USD. 

Con's without being a mere optional alternative would include

Cons
Exposing the sources greater link directory paths as outlined in that request would serve to fingerprint that sources personal location far more than exists now. Internet Service Providers processing a rubber stamped copy-write abuse order by a shyster attorney group often brush that aside due to such demands entirely based on an alleged IPv4 location which can be spoofed, cloned or a relayed proxy. Further providing the same information plus the alleged copy-write infringing users local directory path link would remove the probability of a reasonably likely false positive allegation to a Sextrillion to 1 probability and a physical location check on that customers device itself for confirmation.

There are many thousands of pre-existing .rscollection files now in use on Retroshares File Sharing Network, its likely whoever came up with that suggestion didn't take the time to make 1000's of them for their fellow peers and suddenly those would be worthless if implemented.

If this alternative to existing .rscollection files is implemented, as a 'Optional' alternative which a Retroshare User might chose or enable, OK.  If those choosing that option want to expose a fingerprint on their physical system in greater fashion to anyone that looks at the suggested composition of the export directory link that's their choice and should have at least a bubble text informing them of that. Keep the pre-existing .rscollection file coding in place to continue to support the many thousands of .rscollection files in place and used daily here.

 G10h4ck wrote "_I have been looking at rscollection thing, and it seems to me that we may eventually use .torrent files instead or support both, this would ease inter-operation too as if someone find a .torrent in the internet could try to download it via retroshare instead of having no other option then to use a torrent client, another thing that would be nice is support for magnet links_"

Three years ago I ran multiple tests of providing both .torrent files as well as I created a new file type which I named torrent.magnet files and myself and a handful of established, dedicated Retroshare File Sharing users poured well out over 100,000 active .torrent and torrent.magnet files onto the Retroshare File Sharing network to test its effectiveness for serving as a decentralized .torrent file supplying network. There was opposition to this at that time by Cave and 1-2 others. The tests nonetheless proved the successful, globally spreading those tiny .torrent files downloaded by fellow Retroshare users and also provided a useful discovery of a possible future enhancement to the existing Retroshare File Sharing Network which would be to add a SQ database tie-in which those tiny .torrent and .torrent.magnet files could fill and a more standardized .torrent file sharing and updating on a decentralized network would be largely in place and complete.
